packages/plugins/whisper/src/whisperClient.js

```diff
@@ -0,0 +1,257 @@
+import { __ } from 'embark-i18n';
+const async = require('async');
+const GethMiner = require('./miner');
+const semver = require('semver');
+const constants = require('embark-core/constants');
+
+const DEFAULTS = {
+  "BIN": "geth",
+  "VERSIONS_SUPPORTED": ">=1.8.14",
+  "NETWORK_TYPE": "custom",
+  "NETWORK_ID": 1337,
+  "RPC_API": ['eth', 'web3', 'net', 'debug', 'personal'],
+  "WS_API": ['eth', 'web3', 'net', 'shh', 'debug', 'pubsub', 'personal'],
+  "DEV_WS_API": ['eth', 'web3', 'net', 'shh', 'debug', 'pubsub', 'personal'],
+  "TARGET_GAS_LIMIT": 8000000
+};
+
+class WhisperGethClient {
+
+  static get DEFAULTS() {
+    return DEFAULTS;
+  }
+
+  constructor(options) {
+    this.config = options && options.hasOwnProperty('config') ? options.config : {};
+    this.communicationConfig = options.communicationConfig;
+    this.env = options && options.hasOwnProperty('env') ? options.env : 'development';
+    this.isDev = options && options.hasOwnProperty('isDev') ? options.isDev : (this.env === 'development');
+    this.name = constants.blockchain.clients.geth;
+    this.prettyName = "Go-Ethereum (https://github.com/ethereum/go-ethereum)";
+    this.bin = this.config.ethereumClientBin || DEFAULTS.BIN;
+    this.versSupported = DEFAULTS.VERSIONS_SUPPORTED;
+    this.httpReady = false;
+    this.wsReady = !this.config.wsRPC;
+  }
+
+  isReady(data) {
+    if (data.indexOf('WebSocket endpoint opened') > -1) {
+      this.wsReady = true;
+    }
+    return this.wsReady;
+  }
+
+  /**
+   * Check if the client needs some sort of 'keep alive' transactions to avoid freezing by inactivity
+   * @returns {boolean} if keep alive is needed
+   */
+  needKeepAlive() {
+    return false;
+  }
+
+  commonOptions() {
+    return [];
+  }
+
+  getMiner() {
+    return new GethMiner({datadir: this.config.datadir});
+  }
```

Comment:
Will the whisper node ever really need to mine?

Reply:
Another good catch, removed.